api/types/appserver.go

```diff
@@ -258,6 +258,27 @@ func (s *AppServerV3) Copy() AppServer {
 	return proto.Clone(s).(*AppServerV3)
 }
 
+// Origin returns the origin value of the resource.
+func (s *AppServerV3) Origin() string {
+	return s.Metadata.Origin()
+}
+
+// SetOrigin sets the origin value of the resource.
+func (s *AppServerV3) SetOrigin(origin string) {
+	s.Metadata.SetOrigin(origin)
+}
+
+// GetAllLabels returns resource's labels.
+func (s *AppServerV3) GetAllLabels() map[string]string {
+	return s.Metadata.Labels
+}
+
+// MatchSearch goes through select field values and tries to
+// match against the list of search values.
+func (s *AppServerV3) MatchSearch(values []string) bool {
+	return MatchSearch(nil, values, nil)
```

Comment:
Why not just `return false`?

Reply:
It should only return false if it's provided with search values, otherwise it's always true with empty search vals. I was going to use it similarly to [MatchLabels](https://github.com/gravitational/teleport/blob/e37e52ec81a2cd11eff86b62e33f5a525ed7d87f/lib/auth/auth_with_roles.go#L831)

Reply:
Makes sense to maintain the behavior consistent.